presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -59,36 +60,36 @@
     private final List<Expression> parameters;
     private String updateType;
 
-    private final IdentityHashMap<Table, Query> namedQueries = new IdentityHashMap<>();
+    private final IdentityMap<Table, Query> namedQueries = new IdentityMap<>(LinkedHashMap::new);
 
-    private final IdentityHashMap<Node, Scope> scopes = new IdentityHashMap<>();
-    private final Set<Expression> columnReferences = newIdentityHashSet();
+    private final IdentityMap<Node, Scope> scopes = new IdentityMap<>(LinkedHashMap::new);
```

Comment:
I would be nicer (and simpler) to have `IdentityLinkedHashMap` class defined with empty constructor instead of composite `new IdentityMap<>(LinkedHashMap::new);`